.eslintrc.js

```diff
@@ -369,6 +372,18 @@ module.exports = {
 						message:
 							'--wp-admin-theme-* variables do not support component theming. Use variables from the COLORS object in packages/components/src/utils/colors-values.js instead.',
 					},
+					{
+						selector:
+							'Literal[value=/var.+--wp-components-color-/]', // allow overriding definitions, but not access with var()
+						message:
+							'To ensure proper fallbacks, --wp-components-color-* variables should not be used directly. Use variables from the COLORS object in packages/components/src/utils/colors-values.js instead.',
+					},
+					{
+						selector:
+							'TemplateElement[value.cooked=/var.+--wp-components-color-/]', // allow overriding definitions, but not access with var()
+						message:
+							'To ensure proper fallbacks, --wp-components-color-* variables should not be used directly. Use variables from the COLORS object in packages/components/src/utils/colors-values.js instead.',
+					},
```

Comment:
Don't know if it makes things more or less understandable, but you could squash these rules with a `:matches`...

```suggestion
					{
						selector:
							// allow overriding definitions, but not access with var()
							':matches(Literal[value=/var.+--wp-components-color-/],TemplateElement[value.cooked=/var.+--wp-components-color-/])',
						message:
							'To ensure proper fallbacks, --wp-components-color-* variables should not be used directly. Use variables from the COLORS object in packages/components/src/utils/colors-values.js instead.',
					},
```

Also, is it worth being a bit more specific with the RegExp? It currently matches anything that starts with `var` and ends with `--wp-components-color-`, so (while highly unlikely to happen, I'll grant you!) could hit some false negatives:

```js
// ⛔️ /var.+--wp-components-color-/
const myCss = 'color: var(--wp-components-color-foreground, #FFF);'
// Matches            ^------------------------^
const myCss = 'color: var(--anything, #000); --wp-components-color-background: #FFF;'
// Matches            ^-------------------------------------------^

// ✅ /var\s*\(\s*--wp-components-color-/
const myCss = 'color: var(--wp-components-color-foreground, #FFF);'
// Matches            ^------------------------^
const myCss = 'color: var(--anything, #000); --wp-components-color-background: #FFF;'
// No match
```

Reply:
> you could squash these rules with a `:matches`

Ooh, TIL! That looks cleaner to me too, thanks.

> Also, is it worth being a bit more specific with the RegExp?

I actually wanted to add the parenthesis too, but I couldn't figure out how to do the escaping without Prettier complaining. Turns out I just needed a double backslash. Done 👍